src/rust/engine/fs/store/src/remote.rs

```diff
@@ -203,6 +207,60 @@ impl ByteStore {
     digest: Digest,
     bytes: ByteSource,
   ) -> Result<(), ByteStoreError>
+  where
+    ByteSource: Fn(Range<usize>) -> Bytes + Send + Sync + 'static,
+  {
+    let len = digest.size_bytes;
+
+    let max_batch_total_size_bytes = {
+      let capabilities = self.get_capabilities().await?;
```

Comment:
You already use `double_checked_cell_async` elsewhere... this seems like another good usecase for that?

Reply:
`get_capabilities` already uses `double_checked_cell_async` in its implementation.